cmd/reconciler/ip.go

```diff
@@ -28,21 +63,22 @@ func main() {
 	}
 	if err != nil {
 		_ = logging.Errorf("failed to create the reconcile looper: %v", err)
-		os.Exit(couldNotStartOrphanedIPMonitor)
+		tolerateKnownErrorCasesOrFail(err, couldNotStartOrphanedIPMonitor)
 	}
 
 	cleanedUpIps, err := ipReconcileLoop.ReconcileIPPools(context.Background())
 	if err != nil {
 		_ = logging.Errorf("failed to clean up IP for allocations: %v", err)
-		os.Exit(failedToReconcileIPPools)
+		tolerateKnownErrorCasesOrFail(err, failedToReconcileIPPools)
 	}
+
 	if len(cleanedUpIps) > 0 {
 		logging.Debugf("successfully cleanup IPs: %+v", cleanedUpIps)
 	} else {
 		logging.Debugf("no IP addresses to cleanup")
 	}
 
 	if err := ipReconcileLoop.ReconcileOverlappingIPAddresses(context.Background()); err != nil {
-		os.Exit(failedToReconcileClusterWideIPs)
+		tolerateKnownErrorCasesOrFail(err, failedToReconcileClusterWideIPs)
```

Comment:
To add to my previous point, if the error here is not a known error case, the error will not ever be logged. If we know that the error here is guaranteed to be a known case, this is perfectly fine.

Reply:
This is a really nice catch, thanks! Definitely need an error logged here.